config/management/operational/src/validator_state.rs

```diff
@@ -2,67 +2,119 @@
-            .validator_set(None)?
-            .iter()
-            .any(|vi| vi.account_address() == &owner_account);
+        let validator_infos = client.validator_set(None)?;
 
-        // TODO(khiemngo): consider return early if the validator is not in the set
+        let mut result = VerifyValidatorStateResult {
+            consensus_key_match: None,
+            consensus_key_unique: None,
+            fullnode_network_key_match: None,
+            in_validator_set: None,
+            validator_network_key_match: None,
+        };
 
-        // Fetch the current on-chain config for this operator's owner
+        // Verify if the validator is in the set.
+        result.in_validator_set = Some(
+            validator_infos
+                .iter()
+                .any(|vi| vi.account_address() == &owner_account),
+        );
+
+        // Fetch the current on-chain config for this operator's owner.
+        // Check if the consensus key held in secure storage matches
+        // that registered on-chain.
         let validator_config = client.validator_config(owner_account).and_then(|vc| {
             DecryptedValidatorConfig::from_validator_config_resource(&vc, owner_account)
         })?;
-
         let storage_key = storage.ed25519_public_from_private(CONSENSUS_KEY)?;
-        let consensus_key_match = storage_key == validator_config.consensus_public_key;
+        result.consensus_key_match = Some(storage_key == validator_config.consensus_public_key);
+
+        // Check if the consensus key is unique
+        result.consensus_key_unique = Some(!validator_infos.iter().any(|vi| {
+            vi.account_address() != &owner_account && vi.consensus_public_key() == &storage_key
+        }));
+
+        // Check if the validator network key held in secure storage
+        // matches that registered on-chain.
+        let storage_key = storage.ed25519_public_from_private(VALIDATOR_NETWORK_KEY)?;
+        result.validator_network_key_match = Some(
+            Some(to_x25519(storage_key)?)
+                == validator_config
+                    .validator_network_address
+                    .find_noise_proto(),
+        );
+
+        // Check if the fullnode network key held in secure storage
+        // matches that registered on-chain.
+        let storage_key = storage.ed25519_public_from_private(FULLNODE_NETWORK_KEY)?;
+        result.fullnode_network_key_match = Some(
+            Some(to_x25519(storage_key)?)
+                == validator_config.fullnode_network_address.find_noise_proto(),
+        );
 
-        // TODO(khiemngo): add checks for validator/fullnode network addresses
-        // TODO(khiemngo): add check for key uniqueness
+        Ok(result)
 
-        Ok(VerifyValidatorStateResult {
-            in_validator_set: Some(in_validator_set),
-            consensus_key_match: Some(consensus_key_match),
-        })
+        // TODO(khiemngo): Check if all keys match locally when compared with the validator infos
```

Comment:
As this function grows in the future, I think we could refactor out these checks into helper methods, so the function doesn't become too large. But, let's iterate 😄